src/TrackerSecondary.java

```diff
@@ -0,0 +1,278 @@
+import components.map.Map;
+
+/**
+ * Layered implementations of secondary methods for {@code Map}.
+ */
+public abstract class TrackerSecondary extends Object implements Tracker {
+
+    /**
+     * Sets the value of {@code name} to zero and returns its previous value.
+     *
+     * @param name
+     *            the name of the statistic to reset the value of
+     * @return the original value assosiated with name
+     * @updates this
+     * @requires name is in this
+     * @ensures <pre>
+     * (name, toZero) is in #this  and
+     * (name, 0) is in this
+     * </pre>
+     */
+    @Override
+    public int toZero(String name) {
```

Comment:
You don't need to include the contracts here. They are inherited. You will get a checkstyle warning but you can ignore it.